### PR TITLE
[ES-1043]  added v2/kyc-auth and  /add-verified-claim end-point with therir service method and testcase

### PR DESCRIPTION
**Pull request by kaifk468**

### CODE REVIEW

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/AvailableClaim.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package io.mosip.esignet.mock.identitysystem.dto;
+
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class AvailableClaim {
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/AvailableClaim.java:8]: Rename to VerifiedClaim

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/KycAuthResponseDtoV2.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.mosip.esignet.mock.identitysystem.dto;
+
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class KycAuthResponseDtoV2 extends KycAuthResponseDto{
+
+    private List<AvailableClaim> availableClaims;
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/KycAuthResponseDtoV2.java:10]: Rename to verifiedClaims

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/entity/VerifiedClaim.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+public class VerifiedClaim {
+
+    @Id
+    @Column(name = "id")
+    private String id;
+
+    @Column(name = "individual_id")
+    private String individualId;
+
+    private String claim;
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/entity/VerifiedClaim.java:21]: Why is there no @column annotation on claim field?

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -75,6 +82,9 @@ public class AuthenticationServiceImpl implements AuthenticationService {
     @Value("${mosip.esignet.authenticator.auth-factor.kba.field-language}")
     private String fieldLang;
 
+    @Value("#{${mosip.esignet.mock.authenticator.ida.oidc-claims}}")
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:85]: Lets rename the property as "mosip.mock.ida.identity-openid-claims-mapping"

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -75,6 +82,9 @@ public class AuthenticationServiceImpl implements AuthenticationService {
     @Value("${mosip.esignet.authenticator.auth-factor.kba.field-language}")
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:82]: Actually this key should also be named as "mosip.mock.ida.kba.default.field-language"

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/IdentityServiceImpl.java [hunk 6]
```diff
@@ -54,4 +65,36 @@ public IdentityData getIdentity(String individualId) throws MockIdentityExceptio
 		return identityData;
 	}
 
+	@Override
+	public void addVerifiedClaim(VerifiedClaimRequestDto verifiedClaimRequestDto) throws MockIdentityException {
+		VerifiedClaim verifiedClaim =null;
+		//validateVerifiedClaim(verifiedClaimRequestDto);
+		String idHash= HelperUtil.generateB64EncodedHash(ALGO_SHA3_256, verifiedClaimRequestDto.getIndividualId()+ verifiedClaimRequestDto.getTrustFramework()+ verifiedClaimRequestDto.getClaim());
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/IdentityServiceImpl.java:72]: Are we not checking if the individualId is present in the identity table? And let's use the lowercase of trust framework to generate the hash.

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/util/ErrorConstants.java [hunk 7]
```diff
@@ -31,5 +31,8 @@ public class ErrorConstants {
 	public static final String INVALID_VERSION = "invalid_version";
 	public static final String INVALID_VALUE = "invalid_value";
 	public static final String DUPLICATE_INDIVIDUAL_ID = "duplicate_individual_id";
-
+    public static final String INVALID_CLAIM = "invalid_claim";
+	public static final String INVALID_TRUST_FRAMEWORK = "invalid_trust_framework";
+	public static final String INVALID_VERIFIED_DATE = "invalid_verified_date";
+	;
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/util/ErrorConstants.java:37]: Why is this extra semicolon?

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/AvailableClaim.java (hunk 1 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/AvailableClaim.java:8]: done

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/entity/VerifiedClaim.java (hunk 3 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/entity/VerifiedClaim.java:21]: added

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/KycAuthResponseDtoV2.java (hunk 2 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/KycAuthResponseDtoV2.java:10]: done

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java (hunk 4 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:85]: renamed

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java (hunk 5 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:82]: renamed

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/util/ErrorConstants.java (hunk 7 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/util/ErrorConstants.java:37]: removed

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/IdentityServiceImpl.java (hunk 6 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/IdentityServiceImpl.java]: added individualId check

**Comment by ase-101** [comment]
Not required, changes are taken into https://github.com/mosip/esignet-mock-services/pull/219

**Pull request closed without merging**